pkg/services/ngalert/notifier/file_store.go

```diff
@@ -62,6 +63,46 @@ func (fileStore *FileStore) FilepathFor(ctx context.Context, filename string) (s
 	return fileStore.pathFor(filename), err
 }
 
+// GetFullState returns a base64-encoded string containing the Alertmanager's silences and nflog.
+func (fileStore *FileStore) GetFullState(ctx context.Context) (string, error) {
+	all, err := fileStore.kv.GetAll(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	silences, ok := all[fileStore.orgID]["silences"]
+	if !ok {
+		fileStore.logger.Warn("No silences found in kvstore", "org", fileStore.orgID)
```

Comment:
I don't think JUST logging is right -- instead, I think we should return an error to let the consumer know that we couldn't marshal one of the states we expected.